frontend/src/app/components/ExperimentPage.js

```diff
@@ -18,7 +18,25 @@ import ExperimentPreFeedbackDialog from '../components/ExperimentPreFeedbackDial
 
 const CHANGE_HEADER_ON = 105;
 
+
 export default class ExperimentPage extends React.Component {
+  render() {
+    const { getExperimentBySlug, params } = this.props;
+    const experiment = getExperimentBySlug(params.slug);
```

Comment:
Then again, having that bit in the router means the router has experiment-knowledge. But, I figured that was less of a problem to have the router know how to adapt from slug to experiment to pass into the routed component.